include/carrier/peer_info.h

```diff
@@ -32,57 +32,114 @@ namespace carrier {
 struct CARRIER_PUBLIC PeerInfo {
     PeerInfo() = default;
 
-    explicit PeerInfo(const Blob& id, const Blob& ip, int port)
-        : nodeId(id), sockaddr(ip, port) {}
+    PeerInfo(const Blob& id, const Blob& pid, uint16_t port, const std::string alt, const Blob& sig, int family = AF_INET);
+    PeerInfo(const Blob& id, const uint16_t port, const std::string alt, const Blob& sig, int family = AF_INET);
 
-    explicit PeerInfo(const Id& id, const std::string& ip, int port)
-        : nodeId(id), sockaddr(ip, port) {}
+    PeerInfo(const Id& id, const Id& pid, uint16_t port, const std::string alt, const std::vector<std::uint8_t>& sig, int family = AF_INET);
 
-    explicit PeerInfo(const Id& id, const Blob& ip, int port)
-        : nodeId(id), sockaddr(ip, port) {}
-
-    PeerInfo(const Id& id, const SocketAddress& _addr) noexcept
-        : nodeId(id), sockaddr(_addr)  {}
-
-    PeerInfo(const PeerInfo& ni)
-        : nodeId(ni.nodeId), sockaddr(ni.sockaddr) {}
+    PeerInfo(const PeerInfo& pi);
 
     const Id& getNodeId() const noexcept {
         return nodeId;
     }
-    const SocketAddress& getAddress() const noexcept {
-        return sockaddr;
+
+    const Id& getProxyId() const noexcept {
+        return proxyId;
     }
-    int getPort() const noexcept {
-        return sockaddr.port();
+
+    uint16_t getPort() const noexcept {
+        return port;
     };
 
-    bool isIPv4() const noexcept { return sockaddr.family() == AF_INET;  }
-    bool isIPv6() const noexcept { return sockaddr.family() == AF_INET6; }
+    const std::string& getAlt() const noexcept {
+        return alt;
+    }
 
-    bool operator==(const PeerInfo& other) const {
-        return nodeId == other.nodeId && sockaddr == other.sockaddr;
+    const std::vector<uint8_t>& getSignature() const noexcept {
+        return signature;
     }
+
+    bool isUsedProxy() const noexcept { return usedProxy;  }
+    bool isIPv4() const noexcept { return family == AF_INET;  }
+    bool isIPv6() const noexcept { return family == AF_INET6; }
+
+    bool isValid() const;
+
+    int getFamily() const noexcept { return family; }
```

Comment:
Why need this API "getFamily()",  user only need to check whether it's IPv4 or IPv6(), and we have such APIs afore.